### ISSUE TITLE
Watchdog detects changes half a second later than dsoprea/PyInotify under Linux

### ISSUE THREAD

**Issue by Larivact** [gorakhargosh/watchdog#390]
I am working on a project where the file changes should be detected as soon as possible.
I started out using watchdog since it's crossplatform and conveniently abstracts the filesystem events,
but I noticed that watchdog takes 0.5s longer than [dsoprea/PyInotify](https://github.com/dsoprea/PyInotify) to detect file changes. Is this just the price you have to pay for the abstraction? Or could watchdog be more optimized?

Code I used for benchmarking:
```python
import time
import watchdog.events
import watchdog.observers

class EventHandler(watchdog.events.FileSystemEventHandler):
	def on_any_event(self, event):
		print(time.time())

observer = watchdog.observers.Observer()
observer.schedule(EventHandler(), '.', recursive=True)
observer.start()
input()
```

```python
import time
import inotify.adapters

i = inotify.adapters.InotifyTree(b'.')

for event in i.event_gen():
	if event and 'IN_CLOSE_WRITE' in event[1]:
		print(time.time())
```

**Comment by DonyorM** [comment]
This has to do with how watchdog creates move events. In order to match inotify's `move_from` and `move_to` events to become a single `move` event as offered by watchdog, there needs to be a buffer time to find the matching event. This is set to 0.5 seconds I believe.

**Comment by Larivact** [comment]
Buffering all events when only `moved_from` needs to be matched with `moved_to` seems to be inefficient. Wouldn't it be better to only wait 0.5s when `moved_from` occurs?

I don't get why my testing resulted in a 0.5s delay the timeouts all seem to be set to 1s.
Anyway is there a way to disable the buffering if you don't need the move events to be merged?

**Comment by DonyorM** [comment]
The delay is found in the `inotify_buffer` module with the delay variable being set to 0.5 on line 30. Right now there doesn't seem to be a way to disable it. However, you could just edit the code and set the delay variable to 0.

**Comment by Larivact** [comment]
@DonyorM thanks. I monkey patched it using

    watchdog.observers.inotify_buffer.InotifyBuffer.delay = 0

While this got rid of the delay the move events interestingly still seem to get merged.

**Comment by BoboTiG** [comment]
Closed with 2b2091a25033d15dcfb00ad6b8139e76d1e037ca.
